apps/ios/Shared/Views/Chat/Group/AddGroupMembersView.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  AddGroupMembersView.swift
+//  SimpleX (iOS)
+//
+//  Created by JRoberts on 22.07.2022.
+//  Copyright © 2022 SimpleX Chat. All rights reserved.
+//
+
+import SwiftUI
+import SimpleXChat
+
+struct AddGroupMembersView: View {
+    @EnvironmentObject var chatModel: ChatModel
+    var groupId: Int64
+    @Binding var chatViewSheet: ChatViewSheet?
+    @State private var unaddedContacts: [Contact] = []
+    @State private var selectedContactIds = Set<Int64>()
+
+    var body: some View {
+        VStack {
+            Text("Contacts")
+                .font(.headline)
+                .padding()
+            if (unaddedContacts.isEmpty) {
+                Text("No unadded contacts")
+                    .foregroundColor(.secondary)
+            } else {
+                HStack {
+                    Button {
+                        selectedContactIds.forEach { contactId in
+                            Task {
```

Comment:
what's the reason to create multiple tasks here, instead of iterating inside a single task? I only see downsides.

Reply:
also, forEach should generally not be used other than for view construction - there is `for in` loop for that.